### PR TITLE
Add Gateway Claiming API from TTS

### PR DESCRIPTION
**Pull request by KrishnaIyer**
#### Summary

Add Gateway Claiming API from TTS

#### Changes

- This basically merges the OS parts from https://github.com/TheThingsIndustries/lorawan-stack/pull/2747
- API and CLI are updated

#### Testing

NA

##### Regressions

NA

#### Notes for Reviewers

- For some dumb reason, I had this branch ready when https://github.com/TheThingsIndustries/lorawan-stack/pull/2747 was in review but forgot to create a PR.
- Also pulled changes from https://github.com/TheThingsIndustries/lorawan-stack/pull/2794

#### Checklist

- [ ] Scope: The referenced issue is addressed, there are no unrelated changes.
- [ ] Compatibility: The changes are backwards compatible with existing API, storage, configuration and CLI, according to the compatibility commitments in `README.md` for the chosen target branch.
- [ ] Documentation: Relevant documentation is added or updated.
- [ ] Changelog: Significant features, behavior changes, deprecations and fixes are added to `CHANGELOG.md`.
- [x] Commits: Commit messages follow guidelines in `CONTRIBUTING.md`, there are no fixup commits left.

### CODE REVIEW

File: cmd/ttn-lw-cli/commands/gateways_claim.go [hunk 1]
```diff
@@ -38,23 +41,20 @@ var (
 The claiming procedure transfers ownership of gateways using the Device
 Claiming Server.
 
-Authentication of gateway claiming is by the Gateway EUI and the claim
-authentication code (which is part of the gateway entity itself) stored
-in the Identity Server.
+Gateways need to be authorized for claiming before they can be claimeed.
```
Comment (johanstokking) [inline cmd/ttn-lw-cli/commands/gateways_claim.go]:
```suggestion
Gateways need to be authorized for claiming before they can be claimed.
```

File: cmd/ttn-lw-cli/commands/gateways_claim.go [hunk 2]
```diff
@@ -38,23 +41,20 @@ var (
...
 
-Claim authentication code validity is controlled by the owner of the
-gateway by setting the value and optionally a time window when the
-code is valid. As part of the claiming, the claim authentication code
-is not transferred to the new claimed gateway. This prevents further claim
-requests.
+Authentication of gateway claiming is by the Gateway EUI and the claim
+authentication code. For LoRa Basics Station gateways, this code is validated
+in external gateway management servers. This is currently only supported
+for The Things Indoor Gateways.
```
Comment (johanstokking) [inline cmd/ttn-lw-cli/commands/gateways_claim.go]:
```suggestion
authentication code. For LoRa Basics Station gateways, this code is validated
in external gateway management servers. This is currently only supported
for The Things Indoor Gateways.
```

I don't think this belongs in the CLI description

File: cmd/ttn-lw-cli/commands/gateways_claim.go [hunk 3]
```diff
@@ -38,23 +41,20 @@ var (
...
+entity, stored in the Identity Server.
 
 As part of claiming, you can optionally provide the target Gateway ID and
-Gateway Server Address.
-
-Additionally, for LoRa Basics Station gateways, it is required to provide
-the current auth key (Ex: A The Things Stack API Key or an auth token)
-used by the gateway and the URI of the Target CUPS server to which the
-gateway should connect.
+Gateway Server Address and a frequency plan ID.
```
Comment (johanstokking) [inline cmd/ttn-lw-cli/commands/gateways_claim.go]:
```suggestion
Gateway Server address and a frequency plan ID.
```

File: cmd/ttn-lw-cli/commands/gateways_claim.go [hunk 4]
```diff
@@ -69,26 +69,42 @@ gateway should connect.
...
+			targetCUPSTrustFile, _ := cmd.Flags().GetString("target-cups-trust-file")
 			targetCUPSURI, _ := cmd.Flags().GetString("target-cups-uri")
+			targetFrequencyPlanId, _ := cmd.Flags().GetString("target-frequency-plan-id")
+
+			var targetCUPSTrust []byte
+			if targetCUPSTrustFile != "" {
+				raw, err := ioutil.ReadFile(targetCUPSTrustFile)
+				if err != nil {
+					return errInvalidTargetCUPSTrust.WithCause(err)
+				}
```
Comment (johanstokking) [inline cmd/ttn-lw-cli/commands/gateways_claim.go]: We have file helpers right?

**Review by johanstokking — COMMENTED** [review]
Small things only

File: cmd/ttn-lw-cli/commands/gateways_claim.go (hunk 4 above)
Comment (KrishnaIyer) [reply cmd/ttn-lw-cli/commands/gateways_claim.go]: Ah ok we do have those then. Fixed.

**Review by johanstokking — APPROVED** [review]

**Pull request merged**